### PR TITLE
fix(SfSelect): correct usage of v-model [SFUI2-1300]

### PR DESCRIPTION
**Pull request by Szymon-dziewonski**
# Related issue

https://vsf.atlassian.net/browse/SFUI2-1300

# Scope of work

SfSelect pass wrongly `modelValue`, as `value` prop and then `emits` state via `@onChange` which did not work 
[3e3ba8d0-4b59-47e1-8d5f-1ef1df271afb.webm](https://github.com/vuestorefront/storefront-ui/assets/2566152/d6982f3c-4f3e-4402-9fe3-91b25a9c25a7)
Wrong applying `value` in `sorting` block, it was served as Object when it should be String

Additionally some block used `as` instead of `tag` prop

# Screenshots of visual changes

# Checklist

- [x] Self code-reviewed
- [x] Changes documented
- [x] Semantic HTML
- [ ] SSR-friendly
- [ ] Caching friendly
- [x] a11y for WCAG 2.0 AA
- [ ] examples created
- [ ] blocks created
- [ ] cypress tests created

### CODE REVIEW

File: apps/docs/components/components/select.md [hunk 1]
```diff
@@ -77,15 +79,15 @@ Since this component uses the native `<select>` element, it inherits its accessi
...
-| `value`             | `string`          |            |                                        |
-| `size`              | `SfSelectSize`   | `'base'`          | `'sm'`, `'base'`,`'lg'`                           |
-| `disabled`         | `boolean`         | `false`         |                                        |
-| `invalid`           | `boolean`         | `false`         |                                        |
-| `required`          | `boolean`         | `false`         |                                        |
-| `placeholder`       | `string`          |             |                                        |
-| `wrapperClassName`  | `string`          |             |                                        |
+| Prop name           | Type              | Default value | Possible values                        |
+| ------------------- | ----------------- | ------------- | -------------------------------------- |
+| `modelValue`        | `string`          |               |                                        |
```
Comment (FRSgit) [inline apps/docs/components/components/select.md:84]: This needs to be wrapper into Vue/React comments - in react prop name still is `value`, right?

File: apps/preview/nuxt/pages/showcases/Filters/Color.vue [hunk 2]
```diff
@@ -18,8 +18,9 @@
         <input v-model="selectedValues" :value="value" class="appearance-none peer" type="checkbox" />
         <span
           class="inline-flex items-center justify-center p-1 transition duration-300 rounded-full cursor-pointer ring-1 ring-neutral-200 ring-inset outline-offset-2 outline-secondary-600 peer-checked:ring-2 peer-checked:ring-primary-700 peer-hover:bg-primary-100 peer-[&:not(:checked):hover]:ring-primary-200 peer-active:bg-primary-200 peer-active:ring-primary-300 peer-disabled:cursor-not-allowed peer-disabled:bg-disabled-100 peer-disabled:opacity-50 peer-disabled:ring-1 peer-disabled:ring-disabled-200 peer-disabled:hover:ring-disabled-200 peer-checked:hover:ring-primary-700 peer-checked:active:ring-primary-700 peer-focus:outline"
-          ><SfThumbnail size="sm" :class="value"
-        /></span>
+        >
+          <SfThumbnail size="sm" :class="value" />
```
Comment (FRSgit) [inline apps/preview/nuxt/pages/showcases/Filters/Color.vue:22]: noice

File: packages/sfui/frameworks/vue/components/SfSelect/SfSelect.vue [hunk 3]
```diff
@@ -41,19 +41,16 @@ const emit = defineEmits<{
...
 
-const selected = ref(props.value);
 const chevronRotated = ref(false);
 const { isFocusVisible } = useFocusVisible();
 
-const clickHandler = () => (chevronRotated.value = true);
-const blurHandler = () => (chevronRotated.value = false);
-const keydownHandler = () => (chevronRotated.value = true);
+const onOpen = () => (chevronRotated.value = true);
+const onClose = () => (chevronRotated.value = false);
```
Comment (FRSgit) [inline packages/sfui/frameworks/vue/components/SfSelect/SfSelect.vue:48]: Wondering, could we use `useDisclaimer` here? 😄

File: apps/docs/components/components/select.md (hunk 1 above)
Comment (Szymon-dziewonski) [reply apps/docs/components/components/select.md:84]: good catch thanks

File: packages/sfui/frameworks/vue/components/SfSelect/SfSelect.vue (hunk 3 above)
Comment (Szymon-dziewonski) [reply packages/sfui/frameworks/vue/components/SfSelect/SfSelect.vue:48]: will check this out, and btw its `useDisclosure`

Comment (Szymon-dziewonski) [reply packages/sfui/frameworks/vue/components/SfSelect/SfSelect.vue]: https://github.com/vuestorefront/storefront-ui/pull/2933/commits/c9728e48d1f4e7d0fb36a8dbb4f0a08d8c053a69

File: apps/docs/components/components/select.md (hunk 1 above)
Comment (Szymon-dziewonski) [reply apps/docs/components/components/select.md]: fixed https://github.com/vuestorefront/storefront-ui/pull/2933/files#diff-0d5843efe9d67df3645842294a5aec69bac6135a633c42f9a28e9062971c7f21R84

**Pull request merged**